util/entropy.c

```diff
@@ -16,61 +18,157 @@
  * XXX win32 where there is no concept of /dev/urandom.
  */
 
+struct entropy_read
+{
+	int fd;
+};
+
 /**
- * entropy_read(buf, buflen):
- * Fill the given buffer with random bytes provided by the operating system.
+ * entropy_read_open():
```

Comment:
Are you _sure_ that you want this to be `_open()` and not `_init()`?  I'm imagining a future where every OS has adopted `getentropy()` and we don't need to mess around with file descriptors.

Reply:
FWIW, here's the "add `getentropy()` to POSIX" proposal:
https://www.austingroupbugs.net/view.php?id=1134

Reply:
It could be `init`/`done` if you prefer.  We can't use `free` since that's reserved for functions which can't fail.